src/individual/BattleController_BeforeMove.c

```diff
@@ -0,0 +1,4354 @@
+            LoadBattleSubSeqScript(ctx, ARC_BATTLE_SUB_SEQ, SUB_SEQ_IMMOBILIZED_BY_LOVE);
+            ctx->server_seq_no = CONTROLLER_COMMAND_RUN_SCRIPT;
+            ctx->next_server_seq_no = CONTROLLER_COMMAND_39;
+            ctx->wb_seq_no = BEFORE_MOVE_START;
+            CopyBattleMonToPartyMon(bsys, ctx, ctx->attack_client);
+            ctx->server_status_flag |= BATTLE_STATUS_CHECK_LOOP_ONLY_ONCE;
+            ctx->waza_status_flag |= MOVE_STATUS_NO_MORE_WORK;
+        }
+    }
+}
+
+// void BattleController_CheckSleepAnnouncement(struct BattleSystem *bsys, struct BattleStruct *ctx) {
+//     if (ctx->battlemon[ctx->attack_client].condition & STATUS_SLEEP) {
+//         if (ctx->current_move_index == MOVE_SNORE || ctx->moveNoTemp == MOVE_SLEEP_TALK) {
+//             LoadBattleSubSeqScript(ctx, ARC_BATTLE_SUB_SEQ, SUB_SEQ_SLEEPING);
+//             ctx->next_server_seq_no = ctx->server_seq_no;
+//             ctx->server_seq_no = CONTROLLER_COMMAND_RUN_SCRIPT;
+//             return;
+//         }
+//     }
+// }
+
+// TODO: unique subscript for Stance Change
+void BattleController_CheckStanceChange(struct BattleSystem *bsys, struct BattleStruct *ctx) {
+    if (ctx->battlemon[ctx->attack_client].ability == ABILITY_STANCE_CHANGE && ctx->battlemon[ctx->attack_client].species == SPECIES_AEGISLASH) {
+        ctx->battlerIdTemp = ctx->attack_client;
+        if (ctx->current_move_index == MOVE_KINGS_SHIELD && ctx->battlemon[ctx->attack_client].form_no == 1) {
+            ctx->battlemon[ctx->battlerIdTemp].form_no = 0;
+            BattleFormChange(ctx->battlerIdTemp, ctx->battlemon[ctx->battlerIdTemp].form_no, bsys, ctx, 0);
+            LoadBattleSubSeqScript(ctx, ARC_BATTLE_SUB_SEQ, SUB_SEQ_FORM_CHANGE);
+            ctx->next_server_seq_no = ctx->server_seq_no;
+            ctx->server_seq_no = CONTROLLER_COMMAND_RUN_SCRIPT;
+        } else if (ctx->moveTbl[ctx->current_move_index].power != 0 && ctx->battlemon[ctx->attack_client].form_no == 0) {
+            ctx->battlemon[ctx->battlerIdTemp].form_no = 1;
+            BattleFormChange(ctx->battlerIdTemp, ctx->battlemon[ctx->battlerIdTemp].form_no, bsys, ctx, 0);
+            LoadBattleSubSeqScript(ctx, ARC_BATTLE_SUB_SEQ, SUB_SEQ_FORM_CHANGE);
+            ctx->next_server_seq_no = ctx->server_seq_no;
+            ctx->server_seq_no = CONTROLLER_COMMAND_RUN_SCRIPT;
+        }
+    }
+}
+
+// TODO: Surely this can be simplified? When Mirror Coat is also modernised?
+BOOL BattlerController_RedirectTarget(struct BattleSystem *bsys, struct BattleStruct *ctx) {
+    int side;
+    int battlerId;
+    int battlerIdTarget;
+    int moveType;
+    int maxBattlers;
+    int battlerIdAttacker = ctx->attack_client;
+    int range;
+
+    if (ctx->defence_client == BATTLER_NONE) {
+        return FALSE;
+    }
+
+    if (GetBattlerAbility(ctx, battlerIdAttacker) == ABILITY_NORMALIZE || GetBattlerAbility(ctx, battlerIdAttacker) == ABILITY_MOLD_BREAKER) {
+        return FALSE;
+    }
```

Comment:
check for mold breaker variations here?

Reply:
Not sure, this is vanilla code. I think it implies if the attacker's ability is Normalize or Mold Breaker then redirection abilities won't trigger.